src/Promitor.Integrations.AzureStorageQueue/AzureStorageQueue.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Auth;
+using Microsoft.WindowsAzure.Storage.Queue;
+
+namespace Promitor.Integrations.AzureStorageQueue
+{
+    public class AzureStorageQueue
+    {
+        private readonly ILogger _logger;
+
+        public AzureStorageQueue(ILogger logger)
+        {
+            _logger = logger;
+        }
+
+        public async Task<int> GetQueueSizeAsync(string accountName, string queueName, string sasToken)
+        {
+            var account = new CloudStorageAccount(new StorageCredentials(sasToken), accountName, null, true);
```

Comment:
This is duplicated, let's refactor getting a queue reference into a method which we can re-use